src/sentry/snuba/metrics/query.py

```diff
@@ -341,7 +340,7 @@ def validate_end(self) -> None:
     def validate_granularity(self) -> None:
         # Logic specific to how we handle time series in discover in terms of granularity and interval
         if (
-            self.use_case_key == UseCaseKey.PERFORMANCE
+            self.use_case_id == UseCaseID.TRANSACTIONS
```

Comment:
Why are you not using `get_use_case_key` also here? To be safe and consistent, I would prefer to use it across the layer in order to keep it logically similar to before (ofc now spans will be supported out of the box but I am not sure they would work right away).

Reply:
Sure, I was hesitant to use `get_use_case_key` here because like you said, I was hesitant whether or not span will work right away.

Reply:
@iambriccardo I'm not sure what are you suggesting ( at this level we shouldn't see a `UseCaseKey` anymore, everything should be UseCaseID, if we don't support spans here we should make it explicit).

Reply:
I changed my mind around it, at first I was like, lets convert every incoming `use_case_id` to `use_case_key` in order to keep the logic identical but that will imply that `SPANS` will be implicitly supported, which is not good.

Sorry for the confusion.